Update functions and translations to present and count children in module “unions”

a2line opened the pull request:
fixes #1463 and #1472 + few format/typo fixes in template file

It uses @gunnarhm commit for [n (number)] translations. 
The translation of [having as children] is to be done, so this PR is temporary marked as draft.

gunnarhm committed:
Add translations of numbers in word format for many languages, fix typo in English

* This change in hd/lang/lexicon.txt adds many languages to the list of numbers in word form "n (number)". After this change only Hebrew and Ossetic of the languages currently supported in v7.1.0 are missing in the table.
Solves issue #1472 : [FEATURE REQUEST] Add missing languages in the table converting numbers to words used before list of children

* The English entry in n (number) table in hd/lang/lexicon.txt had twenty-four doubled and double slashes between twenty-three to twenty-nine resulting in erroneous translations for numbers over 22.
Solves issue #1476 : [BUG]Lower number of children (or none) reported for English language in the person view, before the list of children for 23 or more children.

a2line committed:
Use the same count function for the presentation of children in unions

* same function “havingaschildren” in all module (u1-5) support singular/plural translations
* for plural [having as children]1 variable %s is the number of children in full text

gunnarhm commented:
> fixes #1463 and #1472 + few format/typo fixes in template file
> 
> It uses @gunnarhm commit for [n (number)] translations. The translation of [having as children] is to be done, so this PR is temporary marked as draft.

Good idea, this makes it possible to have case and gender and single /plural inflection of the wording. I have prepared parts of the new "having as children" table, and will compose these parts to a complete table for most languages. ( but not checked by language experts ).

gunnarhm commented:
> > fixes #1463 and #1472 + few format/typo fixes in template file
> > It uses @gunnarhm commit for [n (number)] translations. The translation of [having as children] is to be done, so this PR is temporary marked as draft.
> 
> Good idea, this makes it possible to have case and gender and single /plural inflection of the wording. I have prepared parts of the new "having as children" table, and will compose these parts to a complete table for most languages. ( but not checked by language experts ).

However, for many languages, it seems to me to be good to have a phrase meaning "their child:" vs "their two children:" rather than "with" and without the "one" for the single child case. So I will make my proposal for that solution. Earlier today, there was also a discussion indicating that the number was maybe not so desirable since we have no indication if it is the real number in the family or just how many are currently recorded in the base. But I will keep on including the number for >1.

a2line commented:
If one day we have a variable in the database to save that the “family is complete” whatever the number of children is (no posterity case), we can improve how we print the information on that count. The problem with this idea (that is mine!) is that at the moment there are nearly no sources in any country that will attest that a family is “complete”. Various census always miss children. All records are linked to a few individuals and never a whole family. So even if we add this option, I see it only good for contemporaries, but it's oral sources: we'll never have a source for that “family is complete” data. Everybody could check the checkbox to say that the family is complete and mislead the other on the count. The same way we can link wrong children, or duplicate some others…
The idea for me is to have the count printed on top more than to have good translations: the enumeration of the children is linked to its count. It simplifies counts by hand that can be made, moreover, when the number of children is big (I see myself on Geneanet… “1, 2, 3… 17 children…” and you recount them to be sure). It also helps to compare databases on that point. The count printed is not the number of children of the couple, it's the number of known children of the couple listed in this database. I understand it can be misleading, but not more than the list of children itself that already carry it.

gunnarhm committed:
Add missing translations for [having as children]

A new having as children table is provided for hd/lang/lexicon.txt matching the new syntax, with one child/ more children and the number in a parameter %s.
All languages are included, but without any language expert or native language users' check.

a2line commented:
I have cherry-picked the two new translations commits of @gunnarhm and I have closed his PRs #1477 and #1479. I squashed his two commits on [n (number)], it skips a “false conflict” to solve for mergers…

This PR solves #1463 linked to request #1472 and fixes #1476. Thank you gunnarhm!
